### PR TITLE
Migrate notebook CR to kubebuilder V2

### PR DESCRIPTION
**Pull request by lluunn**
Why:
1. simpler structure
1. We can then follow [this](https://book.kubebuilder.io/multiversion-tutorial/tutorial.html) to support multiple version (v1alpha1, v1beta1)

Should not have any functionality change. 
Tested manually (build container, use the new image as the controller, able to get notebook running)

/cc @jlewi @kimwnasptd 

related to https://github.com/kubeflow/kubeflow/issues/3655

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/4013)

### CODE REVIEW

**Comment by kimwnasptd** [comment]
LGTM also from my end. Nice work Lun-Kai

File: components/notebook-controller/config/samples/v1alpha1_notebook.yaml [hunk 1]
```diff
@@ -1,18 +1,7 @@
...
-    spec:
-      containers:
-      - image: "gcr.io/kubeflow-images-public/tensorflow-1.10.1-notebook-cpu:v0.3.0"
-        name: "notebook"
-        resources:
-          requests:
-            cpu: "500m"
-            memory: "1Gi"
-        workingDir: "/home/jovyan"
+  foo: bar
```
[the remark this answers is not in the document]
Comment (jlewi) [reply components/notebook-controller/config/samples/v1alpha1_notebook.yaml]: @swiftdiaries Do you want to file a separate issue for that?

**Comment by lluunn** [comment]
yo @googlebot

**Comment by Gekho457** [comment]
@lluunn Probably a weird place to ask, but what did you do to get a podSpec schema into the notebook CRD?
 
https://github.com/kubeflow/kubeflow/blob/master/components/notebook-controller/config/crd/bases/kubeflow.org_notebooks.yaml#L44

**Pull request merged**